packages/angular_devkit/build_angular/src/builders/browser-esbuild/compiler-plugin.ts

```diff
@@ -0,0 +1,376 @@
+          (result.errors ??= []).push(...errors);
+          (result.warnings ??= []).push(...warnings);
+
+          return contents;
+        };
+
+        resourceHost.transformResource = async function (data, context) {
+          // Only style resources are supported currently
+          if (context.resourceFile || context.type !== 'style') {
+            return null;
+          }
+
+          const { contents, errors, warnings } = await bundleStylesheetData(
+            data,
+            {
+              resolvePath: path.dirname(context.containingFile),
+              virtualName: context.containingFile,
+            },
+            styleOptions,
+          );
+          (result.errors ??= []).push(...errors);
+          (result.warnings ??= []).push(...warnings);
+
+          return { content: contents };
+        };
+
+        // Create the Angular specific program that contains the Angular compiler
+        const angularProgram = new compilerCli.NgtscProgram(
+          rootNames,
+          compilerOptions,
+          host,
+          // this.ngtscNextProgram,
+        );
+        const angularCompiler = angularProgram.compiler;
+        const { ignoreForDiagnostics, ignoreForEmit } = angularCompiler;
+        const typeScriptProgram = angularProgram.getTsProgram();
+
+        const builder = ts.createAbstractBuilder(typeScriptProgram, host);
+
+        // Collect program level diagnostics
+        const diagnostics = [
+          ...angularCompiler.getOptionDiagnostics(),
+          ...builder.getOptionsDiagnostics(),
+          ...builder.getGlobalDiagnostics(),
+        ];
+
+        await angularCompiler.analyzeAsync();
+
+        // Collect source file specific diagnostics
+        const OptimizeFor = compilerCli.OptimizeFor;
+        for (const sourceFile of builder.getSourceFiles()) {
+          if (!ignoreForDiagnostics.has(sourceFile)) {
+            diagnostics.push(...builder.getSyntacticDiagnostics(sourceFile));
+            diagnostics.push(...builder.getSemanticDiagnostics(sourceFile));
+
+            const angularDiagnostics = angularCompiler.getDiagnosticsForFile(
+              sourceFile,
+              OptimizeFor.WholeProgram,
+            );
+            diagnostics.push(...angularDiagnostics);
+          }
+        }
+
+        for (const diagnostic of diagnostics) {
+          const message = convertTypeScriptDiagnostic(diagnostic, host);
+          if (diagnostic.category === ts.DiagnosticCategory.Error) {
+            (result.errors ??= []).push(message);
+          } else {
+            (result.warnings ??= []).push(message);
+          }
+        }
+
+        fileEmitter = createFileEmitter(
+          builder,
+          mergeTransformers(
+            angularCompiler.prepareEmit().transformers,
+            createAotTransformers(builder, {}),
+          ),
+          () => [],
+        );
+
+        return result;
+      });
+
+      build.onLoad({ filter: /\.[cm]?tsx?$/ }, async (args) => {
```

Comment:
Shouldn't this be `/\.[cm]?[tj]sx?$/` when `allowJs` is enabled?

Reply:
Added `allowJs` support